test/e2e/toolbox_test.go

```diff
@@ -0,0 +1,346 @@
+
+		time.Sleep(3 * time.Second)
+
+		session = podmanTest.Podman([]string{"logs", "test"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		session = podmanTest.Podman([]string{"exec", "test", "cat", "/etc/passwd"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		ok, _ := session.GrepString(fmt.Sprintf("%s:x:%s:%s::%s:%s",
+			username,
+			uid,
+			gid,
+			homeDir,
+			shell))
+		Expect(ok).To(BeTrue())
+	})
+
+	It("podman create --userns=keep-id + podman exec - adding group with groupadd", func() {
+		var session *PodmanSessionIntegration
+
+		var groupName string = "testgroup"
+		var gid string = "1001"
+
+		session = podmanTest.Podman([]string{"create", "--name", "test", "--userns=keep-id", "--user", "root:root", fedoraToolbox, "sh", "-c",
+			fmt.Sprintf("groupadd --gid %s %s; sleep 1000",
+				gid, groupName)})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		session = podmanTest.Podman([]string{"start", "test"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		time.Sleep(3 * time.Second)
+
+		session = podmanTest.Podman([]string{"logs", "test"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		session = podmanTest.Podman([]string{"exec", "test", "cat", "/etc/group"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		ok, _ := session.GrepString(groupName)
+		Expect(ok).To(BeTrue())
+	})
+
+	It("podman create --userns=keep-id - entrypoint - modifying existing user with usermod - add to new group, change home/shell/uid", func() {
+		var session *PodmanSessionIntegration
+		var badHomeDir string = "/home/badtestuser"
+		var badShell string = "/bin/sh"
+		var badUID string = "1001"
+		var username string = "testuser"
+		var homeDir string = "/home/testuser"
+		var shell string = "/bin/bash"
+		var uid string = "2000"
+		var groupName string = "testgroup"
+		var gid string = "2000"
+
+		expectedUser := fmt.Sprintf("%s:x:%s:%s::%s:%s",
+			username,
+			uid,
+			gid,
+			homeDir,
+			shell)
+		fmt.Printf("Expected line in /etc/passwd: %s\n", expectedUser)
+
+		session = podmanTest.Podman([]string{"create", "--name", "test", "--userns=keep-id", "--user", "root:root", fedoraToolbox, "sh", "-c",
+			fmt.Sprintf("useradd --home-dir %s --shell %s --uid %s %s; groupadd --gid %s %s; usermod --append --groups wheel --home %s --shell %s --uid %s --gid %s %s; sleep 1000",
+				badHomeDir, badShell, badUID, username, gid, groupName, homeDir, shell, uid, gid, username)})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		session = podmanTest.Podman([]string{"start", "test"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		time.Sleep(3 * time.Second)
+
+		session = podmanTest.Podman([]string{"logs", "test"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
```

Comment:
As long as you're running `logs`, wouldn't it be a good idea to check for error output from the `baduser` add?

Reply:
`baduser` does not imply the invocation of `useradd` should fail. I added a comment that clarifies that fact. Nevertheless, I added checks after the invocation of `podman logs` to make sure it is ok.